lib/screens/home/home_screen.dart

```diff
@@ -23,51 +28,133 @@ class _HomeScreenState extends State<HomeScreen> {
   final HomeController homeController = Get.put(HomeController());
   final TransController transController = Get.put(TransController());
   final UserController userController = Get.find();
+  String updateURL = "";
+  String changelog = "";
   final _tabs = [
     const Transactions(),
     const Categories(),
     AccountsScreen(),
     UserSettings()
   ];
 
+  Widget updateDisplay() {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 10),
+      child: SizedBox(
+        width: double.infinity,
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            const SizedBox(
+              height: 20,
+            ),
+            const Text(
+              "Update Available !!\n",
+              style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
+            ),
+            Text(changelog),
+            const SizedBox(
+              height: 20,
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                TextButton(
+                    onPressed: () {
+                      Utils.openLink(updateURL);
+                      Get.back();
+                    },
+                    child: const Text(
+                      "Download",
+                    )),
+                TextButton(
+                    onPressed: () {
+                      Get.back();
+                    },
+                    child: const Text("No , let it be"))
+              ],
+            )
+          ],
+        ),
+      ),
+    );
+  }
+
+  checkForUpdates(BuildContext context) async {
+    var res = await http.get(Uri.parse(
+        Const.updateCheck));
+
+    if (res.statusCode == 200) {
+      var json = jsonDecode(res.body);
+      if (json == null || res.statusCode != 200) {
+        return null;
+      }
+
+      if (json['version'] > Const.version) {
```

Comment:
lets use package_info_plus to get the build number and use here. we might forget to change this stuff.